servlet/src/main/scala/org/http4s/servlet/AsyncHttp4sServlet.scala

```diff
@@ -54,16 +41,17 @@ class AsyncHttp4sServlet[F[_]](
       ctx.setTimeout(asyncTimeoutMillis)
       // Must be done on the container thread for Tomcat's sake when using async I/O.
       val bodyWriter = servletIo.initWriter(servletResponse)
-      unsafeRunAsync(
-        toRequest(servletRequest).fold(
-          onParseFailure(_, servletResponse, bodyWriter),
-          handleRequest(ctx, _, bodyWriter)
-        )) {
-        case Right(()) =>
-          IO(ctx.complete())
-        case Left(t) =>
-          IO(errorHandler(servletRequest, servletResponse)(t))
-      }
+      F.runAsync(
```

Comment:
This means we are no longer shifting off the servlet thread pool.

Discussion [starts about here](https://gitter.im/http4s/http4s?at=5b81987294f8164c179bf2d2).